docs/user/openstack/install_upi.md

```diff
@@ -0,0 +1,861 @@
+# Installing OpenShift on OpenStack User-Provisioned Infrastructure
+
+The User-Provisioned Infrastructure (UPI) process installs OpenShift in stages, providing opportunities for modifications or integrating with existing infrastructure.
+
+It contrasts with the fully-automated Installer-Provisioned Infrastructure (IPI) which creates everything in one go.
+
+With UPI creating the cloud (OpenStack) resources (e.g. Nova servers, Neutron ports, security groups) is the responsibility of the person deploying OpenShift.
+
+The installer is still used to generate the ignition files and monitor the installation process.
+
+This provides a greater flexibility at the cost of a more explicit and interactive process.
+
+
+## Prerequisites
+
+The requirements for UPI are broadly similar to the [ones for OpenStack IPI][ipi-reqs]:
+
+[ipi-reqs]: ./openstack#openstack-requirements
```

Comment:
This expands to
```
https://github.com/openshift/installer/blob/master/docs/user/openstack/openstack#openstack-requirements
```
...which is broken.

Should instead be:
```
https://github.com/openshift/installer/tree/master/docs/user/openstack#openstack-requirements
```

The sanest way to getting there with a relative link could be:
```
[ipi-reqs]:  ./README.md#openstack-requirements
```

Reply:
or poerhaps `../#openstack-requirements`? I have never tried this one

Reply:
Thanks!